app/controllers/api/plans_base_controller.rb

```diff
@@ -104,38 +108,37 @@ def update
   end
 
   def destroy
-    @plan.destroy
+    plan.destroy
 
     return yield if block_given?
 
-    json = { notice: 'The plan was deleted', id: @plan.id }
+    json = { notice: 'The plan was deleted', id: plan.id }
     respond_to do |format|
       format.json { render json: json, status: :ok }
     end
   end
 
   def plans_index_path
-    polymorphic_path([:admin, @service, collection.build])
+    polymorphic_path([:admin, service, collection.build])
   end
 
-  def assign_plan!(issuer, assoc)
-    plan = !@plan || issuer.send(assoc) == @plan ? nil : @plan
-    issuer.send("#{assoc}=", plan)
+  def assign_plan(issuer, assoc)
```

Comment:
Also by convention there should be a safe without-! counterpart.
The reason a removed these bangs is mainly because `reek` complains with `MissingSafeMethod`. IMHO since these are not public methods and they have not safe version I don't see what value the bang adds.

Quoting from [an article on this topic](https://thecodest.co/blog/ruby-software-development-what-are-the-bang-methods-and-when-to-create-them/):
> There is no point in creating a bang method if its counterpart without an exclamation mark does not exist

Reply:
I'll revert it back to its original name.